drivers/led_strip/ws2812_i2s.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2022 Jonathan Rico
+ *
+ * Adapted from the SPI driver, using the procedure in this blog post:
+ * https://electronut.in/nrf52-i2s-ws2812/
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT worldsemi_ws2812_i2s
+
+#include <string.h>
+
+#include <zephyr/drivers/led_strip.h>
+
+#define LOG_LEVEL CONFIG_LED_STRIP_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ws2812_i2s);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2s.h>
+#include <zephyr/dt-bindings/led/led.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+
+#define WS2812_I2S_PRE_DELAY_WORDS 1
+#define WS2812_I2S_LRCK_HZ	   100000
```

Comment:
I think it would be helpful to have a comment somewhere explaining the relation between the WS clock, the MS clock, and the four bits describing the encoding of the zero and one frames.